benchmark/src/main/java/com/linecorp/decaton/benchmark/BenchmarkResult.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.benchmark;
+
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.time.Duration;
+import java.util.Map.Entry;
+
+import lombok.Value;
+import lombok.experimental.Accessors;
+
+@Value
+@Accessors(fluent = true)
+public class BenchmarkResult {
+    @Value
+    public static class Durations {
+        Duration avg;
```

Comment:
To observe consumer's performance, quantiles are more appropriate I think?

e.g. few outlier can affect avg delivery latency even though most of tasks are delivered in very low latency

Reply:
I thought about that but now I'm thinking percentiles are not that useful for this benchmark.
So the reason we care about delivery latency in this benchmark is by pretty basic restriction in queue processing; if queue contains records 1, 2, 3 and 1 took N seconds to process then that latency appears as the delivery latency of all below - 2 and 3.
So assuming the other parts of delivery latency - e.g, kafka broker's performance for handling Produce/Fetch requests - to be constant, basically the delivery latency in this benchmark will keep increasing from front to end but does not decrease.
We're interested in that "delivery latency as the result of accumulated processing duration" because we want decaton to be able to handle that aspect much better than others, that is, by processing records of different keys in different threads record 2 and 3 will no longer get affected for its delivery latency by 1's exceptionally long processing latency.
For this purpose, I think what we need exactly is just the longest delivery latency observed through the benchmark, which should be the value observed for the last consumed record. 
However, we might have to expect that the maximum latency got affected unexpectedly by something not the above property, for example kafka broker's outlier response time.
I thought that we can distinguish such case having average because if the worst delivery latency was caused by the reason which is different from all others (N-1), then there should be large gap between average and max.
WDYT?

Reply:
Thank you for your explanation.
Sounds reasonable.